logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageMention.kt

```diff
@@ -0,0 +1,50 @@
+package com.wire.kalium.logic.data.message
+
+import com.wire.kalium.logic.data.id.IdMapper
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.persistence.dao.message.MessageEntity
+import com.wire.kalium.protobuf.messages.Mention
+
+data class MessageMention(
+    val start: Int,
+    val length: Int,
+    val userId: UserId?
```

Comment:
Why is the `userId` optional? Because the qualified ID on the proto mention is optional and we don't deal with the fallback to the unqualified ID?

Reply:
According to the [doc](https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/157319329/Use+case+receive+and+display+mentions):
>The user_id field is optional, because in future we might support mentions to everyone. At the moment though it should contain a user ID.

So for now, we only handle direct mentions thus we ignore the ones that doesn't have userId, but we still keep them for the future.

Reply:
Ok I see, I would suggest that we filter out null userIds at the DAO level though so that the logic level can assume that we always have a user IDs.

Reply:
Ok 👍